pyfr/plugins/writer.py

```diff
@@ -58,6 +58,18 @@ def __call__(self, intg):
         else:
             metadata = None
 
+        # Fetch data from other plugins and add it to metadata with ad-hoc keys
+        for csh in intg.completed_step_handlers:
+            try:
+                opdata = csh.serialise(intg)
+
+                if rank == root:
+                    prefix = intg.get_plugin_data_prefix(csh.name, csh.suffix)
+                    metadata.update(
```

Comment:
Just asking, will plugin data always belong to `metadata`? It could get confusing if we store more than _metadata_ (i.e solution fields)

Reply:
Yes, if we want to store more substantial amounts of data an additional mechanism (beyond what the writers already have access to) will be needed, as the metadata variable is written out by the root rank and so does not scale well with large amounts of data.